addonscript/src/addonscript/api.py

```diff
@@ -1,26 +1,36 @@
 """API helpers for addonscript."""
 
+import hashlib
+import os
+
 from aiohttp.client_exceptions import ClientResponseError
 
 from addonscript.exceptions import AMOConflictError, AuthFailedError, AuthInsufficientPermissionsError, FatalSignatureError, SignatureError
 from addonscript.task import get_channel
-from addonscript.utils import amo_download, amo_get, amo_put, get_api_url
+from addonscript.utils import amo_download, amo_get, amo_post, amo_put, get_api_url
+
+# https://addons-server.readthedocs.io/en/latest/topics/api/addons.html#upload-create
+UPLOAD_CREATE = "api/v5/addons/upload/"
 
-# https://addons-server.readthedocs.io/en/latest/topics/api/signing.html#uploading-a-version
-UPLOAD_VERSION = "api/v4/addons/{id}/versions/{version}/"
+# https://addons-server.readthedocs.io/en/latest/topics/api/addons.html#upload-detail
+UPLOAD_DETAIL = "api/v5/addons/upload/{uuid}/"
 
-# https://addons-server.readthedocs.io/en/latest/topics/api/signing.html#checking-the-status-of-your-upload
-UPLOAD_STATUS = "api/v4/addons/{id}/versions/{version}/"
-UPLOAD_STATUS_PK = "api/v4/addons/{id}/versions/{version}/uploads/{upload_pk}/"
+# https://addons-server.readthedocs.io/en/latest/topics/api/addons.html#versions-list
+VERSION_LIST = "api/v5/addons/addon/{id}/versions/?filter=all_with_unlisted"
 
+# https://addons-server.readthedocs.io/en/latest/topics/api/addons.html#version-detail
+VERSION_DETAIL = "api/v5/addons/addon/{id}/versions/{version}/"
 
-# https://addons-server.readthedocs.io/en/latest/topics/api/applications.html
+# https://addons-server.readthedocs.io/en/latest/topics/api/addons.html#put-create-or-edit
+ADDON_CREATE_OR_EDIT = "api/v5/addons/addon/{id}/"
+
+# https://addons-server.readthedocs.io/en/latest/topics/api/v4_frozen/applications.html
 # XXX when we want to support multiple products, we'll need to unhardcode the
 #     `firefox` below
-ADD_VERSION = "api/v4/applications/firefox/{version}/"
+ADD_APP_VERSION = "api/v4/applications/firefox/{version}/"
 
 
-async def add_version(context, version):
+async def add_app_version(context, version):
```

Comment:
At first glance, **_app_** seems misleading, I feel like it should be **_addon_** instead?

Reply:
No this is actually the app (firefox) version. :)